pkg/util/shared/shared.go

```diff
@@ -0,0 +1,24 @@
+// shared package provides utility functions that are shared between the
+// webhooks and the controllers.
+package shared
+
+import (
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func GenJobName(jsName, rjobName string, jobIndex int) string {
+	return fmt.Sprintf("%s-%s-%d", jsName, rjobName, jobIndex)
+}
+
+func IsLeaderPod(pod *corev1.Pod) bool {
```

Comment:
I ended up needing to use this function in the pod reconciler, to check if there is a mismatch between leader and follower pods within a given topology (logic removed from this PR to simplify reviewing), so that's why I placed it in this shared package. 

Yeah naming this package is hard, but I do think we do need a package of shared functions used by both webhooks and controllers, to serve as a single source of truth and avoid code duplication of course.

Reply:
util by definition is a pkg of "shared" functions, may be rename the pkg to `placement` to indicate that those functions are for exclusive placement.

Reply:
you can do that in the followup PR